src/Util.cpp

```diff
@@ -411,6 +412,29 @@ std::string dir_make_temp() {
     #endif
 }
 
+std::vector<char> read_entire(const std::string &pathname) {
+    std::ifstream f(pathname, std::ios::in|std::ios::binary);
+    std::vector<char> result;
+
+    f.seekg(0, std::ifstream::end);
+    size_t size = f.tellg();
+    result.resize(size);
+    f.seekg(0, std::ifstream::beg);
+    f.read(result.data(), result.size());
+    internal_assert(f.good()) << "Unable to read file: " << pathname;
+    f.close();
+    return result;
+}
+
+void write_entire(const std::string &pathname, const void *source, size_t source_len) {
+    std::ofstream f(pathname, std::ios::out|std::ios::binary);
+
+    f.write(reinterpret_cast<const char*>(source), source_len);
+    f.flush();
+    internal_assert(f.good()) << "Unable to write file: " << pathname;
```

Comment:
I think the error checking should be in the caller, so the assert is more meaningful. Same for the read assert, though maybe it means read_entire needs to return a bool and take a reference to a vector as an argument?

Reply:
Not sure about this; I like the idea of a "can't forget to check the error" function for these purposes.

Reply:
Also, the other file utilities in this file tend to have the same assert-fail behavior, rather than returning an error code.